### PR TITLE
Added a readme template for our repos

### PR DESCRIPTION
**Pull request by SkylerAvery**
This is a PR for the Front end guild objective for improving Readmes. I had a look through some of our repos, comparing them with each other and also looking at public repos from numerous organisations. Turns out most examples I saw were very poorly done and hard to follow for newbies to the project/team.

To address this, I have created a readme template that covers the common areas that a readme should cover. These will vary between repos but essential should be `How to run locally`, `Testing`, `Installation`, `Common Questions` and some system overview information, including frequently asked questions.

Let me know what you think, I am currently implementing this template to some of our front end repos so any feedback would be awesome. Cheers, Louis.

### CODE REVIEW

File: readme-template.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+<!-- Not all of these are essential but keep this in mind for the repo you are using -->
+![Build badge]() ![Test Coverage]() ![npm package]()
+
+<!--Short sentence on what the repo does-->
+
+<!-- possibly list experts in this repo for who to go to when encountering a problem-->
+
+##Contents
+* [Further Documentation](#documentation)
```
Comment (cameronviner) [inline readme-template.md]: I think it would be worth moving the location of the documentation after running locally as most of the time I mainly looking on how to set up the repo and get going. 

Do you think running locally needs its own section? 99% of our repos are going to be `npm run start`

File: readme-template.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+<a name="overview"></a>
+## System Overview
+
+* More in-depth overview of the system, what this is trying to accomplish and how it achieves its goal.
+* Keep it concise, write this section like the first paragraph of a blog post, what is the point of this repo?
+
+<a name="installation"></a>
+## Installation
+
+* Recordit of installation setup for new users
```
Comment (cameronviner) [inline readme-template.md]: I'd remove the record it mention here and in the section below, These could quickly become out of date. There is also a chance that we will record something as part of the process that we don't want to be public (certain config vars for example)

**Review by cameronviner — DISMISSED** [review]

File: readme-template.md (hunk 1 above)
Comment (SkylerAvery) [reply readme-template.md]: I dunno, I am in 2 minds with this. I think for the initial run through of the repo yes, but after you have it set up, your're most likely going to be reading other documentation. We can by all means change it later or if more people disagree on the placement we can shift it :)

File: readme-template.md [hunk 3]
```diff
@@ -1,106 +1,93 @@
 # Repo Title
 
-<!-- Not all of these are essential but keep this in mind for the repo you are using -->
-![Build badge]() ![Test Coverage]() ![npm package]()
+<!-- Not all of these are essential but keep this in mind for the repo you are using
+![Build badge]() ![Test Coverage]() ![npm package]() -->
 
 <!--Short sentence on what the repo does-->
 
 <!-- possibly list experts in this repo for who to go to when encountering a problem-->
```
Comment (Smuger07) [inline readme-template.md]: Don't think we need this anymore as we've moved away from it

File: readme-template.md [hunk 4]
```diff
@@ -1,106 +1,93 @@
...
 
 * More in-depth overview of the system, what this is trying to accomplish and how it achieves its goal.
 * Keep it concise, write this section like the first paragraph of a blog post, what is the point of this repo?
+* Perhaps label some of the core tech used in the repo and link to the external websites/documentation
 
-<a name="installation"></a>
 ## Installation
 
-* Recordit of installation setup for new users
 * Step-by-step guide on installation
```
Comment (Smuger07) [inline readme-template.md]: Possibly add "run through this with someone unfamiliar with the repo and see if it's clear to understand" or something similar

File: readme-template.md [hunk 5]
```diff
@@ -1,106 +1,93 @@
...
+* What CI do the test run on?
+* What testing framework is used?
+* How are the tests structured in this repo? Is there a strict writing style to follow?
+
 ### Automated Testing
 
-* Step-by-step to run unit/selenium tests
+* Step-by-step to run all automated testing.
+* What are some of the commands I need to know when running test in my local environment?
+* Any tips on running singular tests or groups of tests at once in this repo?
```
Comment (Smuger07) [inline readme-template.md]: Would be good to link off to the additional automated testing docs as to make the ReadMe with too much information

File: readme-template.md [hunk 6]
```diff
@@ -1,106 +1,93 @@
...
+
 ## Contributing Rules
 
 * Link to HX-culture documentation on contribution
 * Any specific rules for this repo?
 
-<a name="faq"></a>
 ## Common Questions
 
 * Frequently asked questions that will help engineers
```
Comment (Smuger07) [inline readme-template.md]: Possibly good to add in "speak to the relevant guild/pod for any other questions" like we have on the Render and Tripapp ReadME

File: readme-template.md (hunk 4 above)
Comment (SkylerAvery) [reply readme-template.md]: I think that is part of the PR process when the template is adapted to other repos, but good shout

File: readme-template.md (hunk 5 above)
Comment (SkylerAvery) [reply readme-template.md]: I think that is a repo by repo basis, some might be rather straight forward (micro services for instance) whilst others might be rather large so adding in the testing doc would be useful

**Review by Smuger07 — APPROVED** [review]
Happy with this 👍

File: readme-template.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+## Linting
+
+* How to I lint specific file types?
+* Do these run via git hooks
+* Link to linting rules
+
+## Deploying
+
+* How do deploys work for new pull requests?]
```
Comment (Nick-Patrick) [inline readme-template.md]: ] at end of line

File: readme-template.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+* Step-by-step guide of setting up locally
+
+## Script Cheat Sheet
+
+* List scripts used in the repo and a description of what they do
+* Example: `npm run build` := Generates `dist` folder with changes made in `lib`
+
+## Testing
+
+* What CI do the test run on?
```
Comment (Nick-Patrick) [inline readme-template.md:53]: maybe a link to the deployment page?

File: readme-template.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+* What testing framework is used?
+* How are the tests structured in this repo? Is there a strict writing style to follow?
```
Comment (Nick-Patrick) [inline readme-template.md]: should that be in here - do we have any repos which are very different for this? i'd prefer to have a general convention for this (in the tech / test best practices docs)

File: readme-template.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+### Automated Testing
+
+* Step-by-step to run all automated testing.
+* What are some of the commands I need to know when running test in my local environment?
+* Any tips on running singular tests or groups of tests at once in this repo?
+
+### Manual Testing
+
+* Link to external regression checklist document
```
Comment (Nick-Patrick) [inline readme-template.md:66]: do we ever have this?

File: readme-template.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+* If anything particular is setup for this repo, describe how you would manually test a feature branch before going to production
+
+## Built With
+
+* What frameworks are used? Links to framework API docs.
+* Essential node modules used, links to documentation/repo.
+
+## Linting
+
+* How to I lint specific file types?
```
Comment (Nick-Patrick) [inline readme-template.md:76]: is this too much info? can a dev just go into the lint config file instead?

File: readme-template.md (hunk 8 above)
Comment (SkylerAvery) [reply readme-template.md:53]: Perhaps a link to the CI site yes (able to find status of the CI from there for instance). The actual deployment checks should be on each PR

File: readme-template.md (hunk 9 above)
Comment (SkylerAvery) [reply readme-template.md]: This is more on folder structure, what testing utilities should be used for certain file types for example

File: readme-template.md (hunk 10 above)
Comment (SkylerAvery) [reply readme-template.md:66]: We do indeed, on a few repos now and growing ;)

File: readme-template.md (hunk 11 above)
Comment (SkylerAvery) [reply readme-template.md:76]: Perhaps, sections like this should be considered on a repo by repo basis. Some might be straight forward to link to the config file. But some of our repos for instance vary in linting rules so makes sense to highlight the differences here

File: readme-template.md (hunk 10 above)
Comment (Nick-Patrick) [reply readme-template.md:66]: nice

**Review by Nick-Patrick — APPROVED** [review]
